### PR TITLE
test(cloudstorage): Add remaining GCP tests for CloudStorage checks

### PR DESCRIPTION
**Pull request by danibarranqueroo**
### Context

There was some checks without tests for CloudStorage in GCP provider.

### Description

Added new tests for:

- [x] `cloudstorage_bucket_log_retention_policy_lock` check
- [x] `cloudstorage_bucket_public_access` check
- [x] `cloudstorage_bucket_uniform_bucket_level_access` check

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+            ]
+
+            check = cloudstorage_bucket_log_retention_policy_lock()
+            result = check.execute()
+
+            assert len(result) == 1
+            assert result[0].status == "PASS"
+            assert re.search(
+                "Retention Policy with Bucket Lock", result[0].status_extended
+            )
```
Comment (puchy22) [inline tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py]: Could you compare the given str with `Log Sink Bucket {bucket.name} has a Retention Policy with Bucket Lock.` not only the end.

File: tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+            assert result[0].resource_id == "bucket1"
+            assert result[0].resource_name == "bucket1"
+            assert result[0].location == "US"
```
Comment (puchy22) [inline tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py]: Please, could you use the real GCP locations declared in fixtures

File: tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                    name="sink1",
+                    destination="storage.googleapis.com/bucket1",
+                    filter="all",
+                    project_id=GCP_PROJECT_ID,
+                )
+            ]
+
+            from prowler.providers.gcp.services.cloudstorage.cloudstorage_service import (
+                Bucket,
+            )
```
Comment (puchy22) [inline tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py]: If it is possible put the imports together like I did in the above comment.

File: tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+            new=logging_client,
+        ):
+            from prowler.providers.gcp.services.cloudstorage.cloudstorage_bucket_log_retention_policy_lock.cloudstorage_bucket_log_retention_policy_lock import (
+                cloudstorage_bucket_log_retention_policy_lock,
+            )
+
+            cloudstorage_client.project_ids = [GCP_PROJECT_ID]
+            cloudstorage_client.region = "global"
+
+            from prowler.providers.gcp.services.logging.logging_service import Sink
```
Comment (puchy22) [inline tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py]:
```suggestion
            from prowler.providers.gcp.services.logging.logging_service import Sink
            from prowler.providers.gcp.services.cloudstorage.cloudstorage_service import (
                Bucket,
            )
```

File: tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_public_access/cloudstorage_bucket_public_access_test.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            cloudstorage_client.region = "global"
+
+            from prowler.providers.gcp.services.cloudstorage.cloudstorage_service import (
+                Bucket,
+            )
+
+            cloudstorage_client.buckets = [
+                Bucket(
+                    name="bucket1",
+                    id="bucket1",
```
Comment (puchy22) [inline tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_public_access/cloudstorage_bucket_public_access_test.py]: Use different names for `name` and `id` and try to match the most possible with GCP real resources, please

**Review by puchy22 — CHANGES_REQUESTED** [review]
Please apply the changes commented to every test

File: tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_public_access/cloudstorage_bucket_public_access_test.py (hunk 5 above)
Comment (danibarranqueroo) [reply tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_public_access/cloudstorage_bucket_public_access_test.py]: GCP resources use the same name and ID, so I'll change both to match the format of real names.

File: tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py (hunk 3 above)
Comment (danibarranqueroo) [reply tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py]: Done

File: tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py (hunk 2 above)
Comment (danibarranqueroo) [reply tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py]: Done

File: tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py (hunk 1 above)
Comment (danibarranqueroo) [reply tests/providers/gcp/services/cloudstorage/cloudstorage_bucket_log_retention_policy_lock/cloudstorage_bucket_log_retention_policy_lock_test.py]: Done

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
This is awesome @danibarranqueroo , congrats on your first PR!
But please, instead of using `re.search`, always assert the whole string. Could you change it for the status_extended?
The rest of it is perfect 😄

**Comment by danibarranqueroo** [comment]
> This is awesome @danibarranqueroo , congrats on your first PR! But please, instead of using `re.search`, always assert the whole string. Could you change it for the status_extended? The rest of it is perfect 😄

Thank you @sergargar, I've just made the changes that you suggested :tada:

**Comment by danibarranqueroo** [comment]
> > This is awesome @danibarranqueroo , congrats on your first PR! But please, instead of using `re.search`, always assert the whole string. Could you change it for the status_extended? The rest of it is perfect 😄
> 
> Thank you @sergargar, I've just made the changes that you suggested 🎉

Since changing that something failed, I'll fix it now and come back when it's done

**Review by MrCloudSec — APPROVED** [review]
👏🏼 👏🏼 👏🏼

**Review by puchy22 — APPROVED** [review]
Awesome job!!! 🚀 🚀

**Pull request merged**